prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt

```diff
@@ -68,36 +70,75 @@ class FHIRTranslator(
                 val receiver = settings.findReceiver(message.receiverFullName)
                     ?: throw RuntimeException("Receiver with name ${message.receiverFullName} was not found")
                 actionHistory.trackActionReceiverInfo(receiver.organizationName, receiver.name)
-                val bodyBytes = getByteArrayFromBundle(receiver, bundle)
+
+                var nextAction = Event.EventAction.BATCH
+                var externalName: String? = null
+                var queueMessage: ReportEventQueueMessage? = null
+                val bodyBytes = if (message.topic.isSendOriginal) {
+                    nextAction = Event.EventAction.SEND
+                    val originalReport = getOriginalReport(message.reportId)
+                    externalName = originalReport.externalName
+                    queueMessage = ReportEventQueueMessage(nextAction, false, message.reportId, "")
```

Comment:
Nit: I think this can be simplified by not instantiating this here?  Seems a little unsafe since `message.reportId` is the wrong value right?

Reply:
Yes wrong value that gets updated below (will fix in #13232)